### ISSUE TITLE
New Problem: Early stopping

### ISSUE THREAD

**Issue by moe18** [Open-Deep-ML/DML-OpenProblem#237]
Early stopping is a regularization technique in machine learning used to prevent overfitting during training. It involves monitoring the performance of a model on a validation set during training.

If the model's performance on the validation set stops improving for a specified number of iterations (patience), training is halted, even if the model's performance on the training data continues to improve. This ensures the model doesn't overfit to the training data and maintains better generalization to unseen data.

**Comment by emharsha1812** [comment]
I will work on this!

**Comment by emharsha1812** [comment]
hi @moe18 
https://github.com/Open-Deep-ML/DML-OpenProblem/pull/389
request you to review
